internal/license/license.go

```diff
@@ -0,0 +1,142 @@
+package license
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/go-resty/resty/v2"
+	"github.com/google/licenseclassifier"
+	"github.com/mholt/archiver/v3"
+)
+
+// Checker is exported
+type Checker struct {
+	classifier *licenseclassifier.License
+	restClient *resty.Client
+}
+
+// NewChecker is exported
+func NewChecker() (*Checker, error) {
+	license, err := licenseclassifier.New(licenseclassifier.DefaultConfidenceThreshold)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Checker{
+		classifier: license,
+		restClient: resty.New(),
+	}, nil
+}
+
+// Type is exportesd
+func (f *Checker) Type(moduleName, version string) (string, error) {
+	v, err := f.getVersion(moduleName, version)
+	if err != nil {
+		return "", err
+	}
+
+	response, err := f.restClient.R().
+		Get(createGoProxyURLForVersion(moduleName, v))
+	if err != nil {
+		return "", err
+	}
+
```

Comment:
In case of response status > 200 OK the stdout message is `opening zip reader: zip: not a valid zip file`.

Test case:
go run cmd/gomodctl/gomodctl.go license github.com/beatlabs/patron v1.30.0

You can do sth like :

	if response.RawResponse.Status != "200 OK" {
		return "Invalid Library.", nil
	}

Reply:
Good catch, I will refactor that part.